### DISCUSSION TITLE
Upload image together with message

### DISCUSSION THREAD

**Discussion by renja-g** [api-platform/core#5193] [Q&A]
Hi, right now I have an endpoint where I can upload images (following the [file-upload-guide](https://api-platform.com/docs/core/file-upload/))
I also have an endpoint where I can create a message, a I have also made a relation to the image entity.
But If I now would like to send a message with an Image, I would have to make two requests.

Is there a way how I can make an endpoint that is wrapping the two endpoints together or what would you recommend?

**Comment by alanpoulain** [comment]
Hello,
You need to create a resource containing both the message and the image and follow this guide: https://api-platform.com/docs/core/file-upload/#uploading-to-an-existing-resource-with-its-fields.

**Reply by renja-g** [reply]
At the end I would have:
`Message` entity (json)
`Image` entity (multipart/form-data)
and
`MessageImage` entity (multipart/form-data)
right?

**Reply by alanpoulain** [reply]
That's right. The `MessageImage` resource can also be retrieved in JSON(-LD).

**Reply by renja-g** [reply]
But don't I need from-data to upload a file?

**Reply by alanpoulain** [reply]
Yes you need it. But you can also add JSON(-LD) (not for creation).
